src/main/webapp/app/exercises/shared/result/result.service.ts

```diff
@@ -94,14 +94,34 @@ export class ResultService implements IResultService {
         const relativeScore = roundValueSpecifiedByCourseSettings(result.score!, getCourseFromExercise(exercise));
         const points = roundValueSpecifiedByCourseSettings((result.score! * exercise.maxPoints!) / 100, getCourseFromExercise(exercise));
         if (exercise.type !== ExerciseType.PROGRAMMING) {
+            if (Result.isAthenaAIResult(result)) {
+                return this.getResultStringNonProgrammingExerciseWithAIFeedback(result, relativeScore, points, short);
+            }
             return this.getResultStringNonProgrammingExercise(relativeScore, points, short);
         } else {
             return this.getResultStringProgrammingExercise(result, exercise as ProgrammingExercise, relativeScore, points, short);
         }
     }
 
     /**
-     * Generates the result string for a programming exercise. Contains the score and points
+     * Generates the result string for a text exercise. Contains the score and points
+     * @param result the result object
+     * @param relativeScore the achieved score in percent
+     * @param points the amount of achieved points
+     * @param short flag that indicates if the resultString should use the short format
+     */
+    private getResultStringNonProgrammingExerciseWithAIFeedback(result: Result, relativeScore: number, points: number, short: boolean | undefined): string {
+        let aiFeedbackMessage: string = '';
+        if (result && Result.isAthenaAIResult(result) && result.successful === undefined) {
+            return this.translateService.instant('artemisApp.result.resultString.automaticAIFeedbackInProgress');
+        } else {
+            aiFeedbackMessage = this.getResultStringNonProgrammingExercise(relativeScore, points, short);
+        }
+        return aiFeedbackMessage + ' (' + this.translateService.instant('artemisApp.result.preliminary') + ')';
```

Comment:
```suggestion
        aiFeedbackMessage = this.getResultStringNonProgrammingExercise(relativeScore, points, short);
        return `${aiFeedbackMessage} (${this.translateService.instant('artemisApp.result.preliminary')})`;
```
Or move all of it into the else branch, up to you...